packages/jest-util/src/fake_timers.js

```diff
@@ -9,537 +9,123 @@
-  clearInterval(intervalId?: number): void,
-  clearTimeout(timeoutId?: number): void,
-  nextTick: (callback: Callback) => void,
-
-  /**
-   * The additional arguments in the following methods are passed to the
-   * callback and thus we don't know their types ahead of time as they can be
-   * anything, which  is why we are disabling the flowtype/no-weak-types rule
-   * here.
-   */
-
-  /* eslint-disable flowtype/no-weak-types */
-  setImmediate(callback: Callback, ms?: number, ...args: Array<any>): number,
-  setInterval(callback: Callback, ms?: number, ...args: Array<any>): number,
-  setTimeout(callback: Callback, ms?: number, ...args: Array<any>): number,
-  /* eslint-enable flowtype/no-weak-types */
-};
-
-type TimerConfig<Ref> = {|
-  idToRef: (id: number) => Ref,
-  refToId: (ref: Ref) => ?number,
-|};
-
-const MS_IN_A_YEAR = 31536000000;
-
-export default class FakeTimers<TimerRef> {
-  _cancelledImmediates: {[key: TimerID]: boolean};
-  _cancelledTicks: {[key: TimerID]: boolean};
+export default class FakeTimers {
+  _clock: Clock;
   _config: ProjectConfig;
-  _disposed: boolean;
-  _fakeTimerAPIs: TimerAPI;
+  _fakingTime: boolean;
   _global: Global;
-  _immediates: Array<Tick>;
+  _lolex: lolex;
   _maxLoops: number;
-  _moduleMocker: ModuleMocker;
-  _now: number;
-  _ticks: Array<Tick>;
-  _timerAPIs: TimerAPI;
-  _timers: {[key: TimerID]: Timer};
-  _uuidCounter: number;
-  _timerConfig: TimerConfig<TimerRef>;
 
   constructor({
     global,
-    moduleMocker,
-    timerConfig,
     config,
     maxLoops,
   }: {
     global: Global,
-    moduleMocker: ModuleMocker,
-    timerConfig: TimerConfig<TimerRef>,
     config: ProjectConfig,
     maxLoops?: number,
   }) {
     this._global = global;
-    this._timerConfig = timerConfig;
     this._config = config;
     this._maxLoops = maxLoops || 100000;
-    this._uuidCounter = 1;
-    this._moduleMocker = moduleMocker;
 
-    // Store original timer APIs for future reference
-    this._timerAPIs = {
-      clearImmediate: global.clearImmediate,
-      clearInterval: global.clearInterval,
-      clearTimeout: global.clearTimeout,
-      nextTick: global.process && global.process.nextTick,
-      setImmediate: global.setImmediate,
-      setInterval: global.setInterval,
-      setTimeout: global.setTimeout,
-    };
-
-    this.reset();
-    this._createMocks();
-
-    // These globally-accessible function are now deprecated!
-    // They will go away very soon, so do not use them!
-    // Instead, use the versions available on the `jest` object
-    global.mockRunTicksRepeatedly = this.runAllTicks.bind(this);
-    global.mockRunTimersOnce = this.runOnlyPendingTimers.bind(this);
-    global.mockAdvanceTimersByTime = this.advanceTimersByTime.bind(this);
-    global.mockRunTimersRepeatedly = this.runAllTimers.bind(this);
-    global.mockClearTimers = this.clearAllTimers.bind(this);
-    global.mockGetTimersCount = () => Object.keys(this._timers).length;
```

Comment:
You're removing `global.mockGetTimersCount` and there isn't an alternative public API for it. Can you add it as a method in the `jest` object? We still use it at Facebook.

Reply:
Added `getTimersCount` in 266a3c3e66329b093cf6c87394d6dee33f366a10